dashboard/app/controllers/scripts_controller.rb

```diff
@@ -5,10 +5,11 @@ class ScriptsController < ApplicationController
   before_action :require_levelbuilder_mode_or_test_env, only: [:edit, :update, :new, :create]
   before_action :authenticate_user!, except: [:show, :vocab, :resources, :code, :standards]
   check_authorization
-  before_action :set_unit, only: [:show, :vocab, :resources, :code, :standards, :edit, :update, :destroy]
+  before_action :set_unit_by_name, only: [:show, :vocab, :resources, :code, :standards, :edit]
   before_action :render_no_access, only: [:show]
   before_action :set_redirect_override, only: [:show]
-  authorize_resource class: 'Unit'
+  authorize_resource class: 'Unit', except: [:update, :destroy]
+  load_and_authorize_resource class: 'Unit', only: [:update, :destroy]
```

Comment:
@davidsbailey , it looks like this change introduced a regression as delete scripts API call uses the script name, and this is resulting in a look up by ID. I have a PR going to fix the destroy and update actions (https://github.com/code-dot-org/code-dot-org/pull/59426)

Can you help me understand why you made this change to split the flow for update and destroy actions.

Reply:
@vijayamanohararaj apologies, I must not have tested the destroy route. I made the overall change in order to be able to deprecate the feature where we show scripts by id, and in the process i had to categorize all actions based on whether they took a name or id.

Reply:
can you say more specifically what is broken in levelbuidler? a screenshot or video would be helpful.

Reply:
I'm not sure I am following the PR you linked to. is there a file missing from it? 

in any case, ideally, update and destroy would take an id. I'm not sure I see any reason they should take a name.

Reply:
Here is the repro,

On levelbuilder scripts page, delete any script, click yes on the confirmation dialog
Actual:
We see below exception
![image](https://github.com/code-dot-org/code-dot-org/assets/124813947/06eb0b14-2481-4ff2-8dfb-60ff71f3e6d8)

>in any case, ideally, update and destroy would take an id. I'm not sure I see any reason they should take a name.

This is making me think that we expect the deletion to use ID and not the name. Sounds like that should be the fix in that case.

The PR right now is only a set of unit tests to assert the expected behavior, so doesn't have any changes yet. From what you are saying, it looks like the set of tests I was adding to change id to name for destroy and update should be reverted, correct?

Reply:
that makes sense. nice TDD!

yes, it seems like delete and update should both use the id, not the name. this makes it easier to use cancan to load_and_authorize_resource.

Reply:
Perfect, thanks for the confirmation, I'll work on that fix.